### PR TITLE
Make logger instances vars in linux

### LINKED ISSUE #9908: Development branch: Linux portable version does not start
### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

Linux

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

Was working a few days ago. Does not work today.

1. download https://builds.jabref.org/main/JabRef-5.10-portable_linux.tar.gz
2. gunzip it
3. launch Jabref (e.g. `~/test/JabRef/bin/JabRef`)

JabRef does not start.
Console message:
````
Exception in thread "main" java.lang.UnsupportedOperationException: Configuration cannot be changed after applying to tinylog
	at org.tinylog.api@2.6.2/org.tinylog.configuration.Configuration.set(Unknown Source)
	at org.jabref@5.10.891/org.jabref.cli.Launcher.lambda$addLogToDisk$0(Unknown Source)
	at java.base/java.lang.Iterable.forEach(Unknown Source)
	at org.jabref@5.10.891/org.jabref.cli.Launcher.addLogToDisk(Unknown Source)
	at org.jabref@5.10.891/org.jabref.cli.Launcher.main(Unknown Source)

````

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #9908

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by HoussemNasri** [comment]
Runs flawlessly on OpenSUSE Leap 15, however, I got this exception, don't know if it's relevant. 
``` bash
2023-05-16 19:33:29 [JavaFX Application Thread] org.jabref.preferences.JabRefPreferences.getUser()
ERROR: Hostname not found. Please go to https://docs.jabref.org/ to find possible problem resolution: java.net.UnknownHostException: localhost.localdomain: localhost.localdomain: Name or service not known
	at java.base/java.net.InetAddress.getLocalHost(InetAddress.java:1781)
	at org.jabref@100.0.0/org.jabref.preferences.JabRefPreferences.getUser(JabRefPreferences.java:2023)
	at org.jabref@100.0.0/org.jabref.preferences.JabRefPreferences.getInternalPreferences(JabRefPreferences.java:1995)
	at org.jabref@100.0.0/org.jabref.preferences.JabRefPreferences.getFilePreferences(JabRefPreferences.java:2116)
	at org.jabref@100.0.0/org.jabref.preferences.JabRefPreferences.getLayoutFormatterPreferences(JabRefPreferences.java:1124)
	at org.jabref@100.0.0/org.jabref.preferences.JabRefPreferences.lambda$getPreviewLayouts$118(JabRefPreferences.java:2353)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.jabref@100.0.0/org.jabref.preferences.JabRefPreferences.getPreviewLayouts(JabRefPreferences.java:2357)
	at org.jabref@100.0.0/org.jabref.preferences.JabRefPreferences.getPreviewPreferences(JabRefPreferences.java:2313)
Hostname not found. Please go to https://docs.jabref.org/ to find possible problem resolution: java.net.UnknownHostException: localhost.localdomain: localhost.localdomain: Name or service not known

	at org.jabref@100.0.0/org.jabref.gui.JabRefFrame.createMenu(JabRefFrame.java:814)
	at org.jabref@100.0.0/org.jabref.gui.JabRefFrame.initLayout(JabRefFrame.java:502)
	at org.jabref@100.0.0/org.jabref.gui.JabRefFrame.init(JabRefFrame.java:648)
	at org.jabref@100.0.0/org.jabref.gui.JabRefGUI.openWindow(JabRefGUI.java:79)
	at org.jabref@100.0.0/org.jabref.gui.JabRefGUI.<init>(JabRefGUI.java:59)
	at org.jabref@100.0.0/org.jabref.gui.MainApplication.start(MainApplication.java:31)
	at javafx.graphics@20/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:839)
	at javafx.graphics@20/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:483)
	at javafx.graphics@20/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at javafx.graphics@20/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
	at javafx.graphics@20/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at javafx.graphics@20/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@20/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.net.UnknownHostException: localhost.localdomain: Name or service not known
	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:52)
	at java.base/java.net.InetAddress$PlatformResolver.lookupByName(InetAddress.java:1059)
	at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1673)
	at java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:1003)
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1663)
	at java.base/java.net.InetAddress.getLocalHost(InetAddress.java:1776)
	... 29 more
Caused by: java.net.UnknownHostException: localhost.localdomain: Name or service not known
` ` `

File: src/main/java/org/jabref/gui/desktop/os/DefaultDesktop.java [hunk 1]
```diff
@@ -8,12 +8,10 @@
 import org.jabref.architecture.AllowedToUseAwt;
 import org.jabref.gui.DialogService;
 
-import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 @AllowedToUseAwt("Requires AWT to open a file")
 public class DefaultDesktop implements NativeDesktop {
-    private static final Logger LOGGER = LoggerFactory.getLogger(NativeDesktop.class);
```
Comment (HoussemNasri) [inline src/main/java/org/jabref/gui/desktop/os/DefaultDesktop.java:16]: See below

**Review by HoussemNasri — APPROVED** [review]
LGTM!

File: src/main/java/org/jabref/gui/desktop/os/Linux.java [hunk 2]
```diff
@@ -20,30 +20,27 @@
 import org.jabref.gui.util.StreamGobbler;
 import org.jabref.logic.l10n.Localization;
 
-import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 @AllowedToUseAwt("Requires AWT to open a file with the native method")
 public class Linux implements NativeDesktop {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(Linux.class);
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/desktop/os/Linux.java:29]: Indeed a good idea!

**Comment by Siedlerchr** [comment]
Thanks for testing. The other exceptions comes from 
` userName = get(DEFAULT_OWNER) + '-' + InetAddress.getLocalHost().getHostName(); `

and indeed it seems to be not that an easy option to get the hostname.
One could add a check InetAddress.getLocalHost().isLoopbackAddress()  or in case of the exception try again with 
and then do a fallback to call the hostname command
https://stackoverflow.com/questions/7348711/recommended-way-to-get-hostname-in-java

**Pull request merged**